Source/ACE.Server/Factories/LootGenerationFactory.cs

```diff
@@ -1545,78 +1488,108 @@ private static int GetWorkmanship(int tier)
+            else if (maxSpellLevel == 4)
+                maxSpellDiff = 150;
+            else if (maxSpellLevel == 5)
+                maxSpellDiff = 200;
+            else if (maxSpellLevel == 6)
+                maxSpellDiff = 250;
+            else if (maxSpellLevel == 7)
+                maxSpellDiff = 300;
+            else if (maxSpellLevel == 8)
+                maxSpellDiff = 400;  
+
+            var tItemSpellCraft = maxSpellDiff * ThreadSafeRandom.Next(minItemSpellCraftRange, maxItemSpellCraftRange);
+
+            if (tItemSpellCraft < 0)
+                tItemSpellCraft = 0;
+
+            int finalItemSpellCraft = (int)Math.Floor(tItemSpellCraft);
+
+            return finalItemSpellCraft;
+
         }
 
-        private static int GetDifficulty(int tier, int spellcraft)
+        private static int GetDifficulty(WorldObject wo, int tier, int itemspellcraft)
         {
-            int difficulty = 0;
-            switch (tier)
-            {
-                case 1:
-                    difficulty = spellcraft + (ThreadSafeRandom.Next(0, 10) * ThreadSafeRandom.Next(1, 3));
-                    break;
-                case 2:
-                    difficulty = spellcraft + (ThreadSafeRandom.Next(0, 10) * ThreadSafeRandom.Next(1, 3));
-                    break;
-                case 3:
-                    difficulty = spellcraft + (ThreadSafeRandom.Next(0, 10) * ThreadSafeRandom.Next(1, 3));
-                    break;
-                case 4:
-                    difficulty = spellcraft + (ThreadSafeRandom.Next(0, 10) * ThreadSafeRandom.Next(1, 3));
-                    break;
-                case 5:
-                    difficulty = spellcraft + (ThreadSafeRandom.Next(0, 10) * ThreadSafeRandom.Next(1, 3));
-                    break;
-                case 6:
-                    difficulty = spellcraft + (ThreadSafeRandom.Next(0, 10) * ThreadSafeRandom.Next(1, 3));
-                    break;
-                case 7:
-                    difficulty = spellcraft + (ThreadSafeRandom.Next(0, 10) * ThreadSafeRandom.Next(1, 3));
-                    break;
-                case 8:
-                    difficulty = spellcraft + (ThreadSafeRandom.Next(0, 10) * ThreadSafeRandom.Next(1, 3));
-                    break;
-                default:
-                    break;
-            }
+            int wieldReq = 1;
+            int sc = 0;  
+            int spc = 1;
+            int epicAddon = 0;
+            int legAddon = 0;
+
+            spc = wo.Biota.PropertiesSpellBook.Count();
+
+            if (wo.EpicCantrips.Count > 0)
+                epicAddon = ThreadSafeRandom.Next(1, 5) * wo.EpicCantrips.Count;
+            if (wo.LegendaryCantrips.Count > 0)
+                legAddon = ThreadSafeRandom.Next(5, 10) * wo.LegendaryCantrips.Count;
+
+            if (wo.ItemAllegianceRankLimit.HasValue)
+                sc = wo.ItemAllegianceRankLimit.Value;
+            if (wo.WieldDifficulty.HasValue)
```

Comment:
this if statement should have { } brackets around it

the way it is currently written is very scary, and very prone to error. i honestly don't even know how the compiler would evaluate this one, without stepping through it in the debugger